fluid-project/hearth
# feat: implement two-factor authentication behind a flag (resolves #3, resolves #23)

- [x] Confirm password before enabling or disabling two-factor authentication
- [x] Display QR code and recovery codes when two-factor authentication has been enabled
- [x] Allow a user to enter either their two-factor authentication code _or_ a recovery code when logging in
- [x] Allow a user to regenerate their recovery codes
- [x] Add a generic PasswordConfirmation modal component

### Testing

1. Install this branch with Composer.
2. Run the Hearth installer using the `--two-factor` flag: `php artisan hearth:install --two-factor`
3. Verify that a user can enable two-factor authentication and configure an authenticator app using a QR code.
4. Verify that a user can log in using an authenticator app.
5. Verify that a user can log in with a recovery code.
6. Verify that a user can regenerate their recovery codes.
7. Verify that a user can disable two-factor authentication.

### Additional Notes

Need to address this: https://github.com/laravel/fortify/issues/201
